hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadata.java

```diff
@@ -34,16 +35,19 @@
  * @param payerKey payer key required to sign the transaction. It is null if payer is missing
  * @param requiredNonPayerKeys list of keys that are required to sign the transaction, in addition
  *     to payer key
+ * @param readKeys the keys that were read during pre-handle
  */
 public record SigTransactionMetadata(
         @NonNull TransactionBody txnBody,
         @NonNull AccountID payer,
         ResponseCodeEnum status,
         @Nullable HederaKey payerKey,
-        List<HederaKey> requiredNonPayerKeys)
+        List<HederaKey> requiredNonPayerKeys,
+        List<ReadKeys> readKeys)
```

Comment:
`ReadKeys` should probably be renamed to signify its the entity numbers, to not confuse with cryptographic keys